go/vt/tabletserver/tabletserver.go

```diff
@@ -233,7 +240,20 @@ func (tsv *TabletServer) StartService(target querypb.Target, dbconfigs dbconfigs
 	if err != nil {
 		return err
 	}
-	return tsv.SetServingType(tabletType, true)
+	return tsv.SetServingType(tabletType, true, nil)
+}
+
+// EnterLameduck causes tabletserver to enter the lameduck state. This
+// state causes health checks to fail, but the behavior of tabletserver
+// otherwise remains the same. Any subsequent calls to SetServingType will
+// cause the tabletserver to exit this mode.
+func (tsv *TabletServer) EnterLameduck() {
+	tsv.lameduck.Set(1)
+}
```

Comment:
If you look at SetServingType, it doesn't broadcast health. I believe tablet manager is responsible for this. We should follow the same principle here also.